webui/react/src/components/ColumnPickerMenu.tsx

```diff
@@ -95,18 +99,28 @@ const ColumnPickerTab: React.FC<ColumnTabProps> = ({
   }, [searchString, totalColumns, tab]);
 
   const allFilteredColumnsChecked = useMemo(() => {
-    return filteredColumns.every((col) => columnState.includes(col.column));
+    return filteredColumns.every((col) => columnState.includes(formatColumnKey(col)));
   }, [columnState, filteredColumns]);
 
   const handleShowHideAll = useCallback(() => {
-    const filteredColumnMap: Record<string, boolean> = filteredColumns.reduce(
-      (acc, col) => ({ ...acc, [col.column]: columnState.includes(col.column) }),
-      {},
-    );
+    const filteredColumnMap: Record<string, boolean> = filteredColumns.reduce((acc, col) => {
+      if (col.location === V1LocationType.RUNMETADATA)
+        return { ...acc, [formatColumnKey(col)]: columnState.includes(formatColumnKey(col)) };
+
+      return { ...acc, [col.column]: columnState.includes(col.column) };
+    }, {});
 
     const newColumns = allFilteredColumnsChecked
       ? columnState.filter((col) => !filteredColumnMap[col])
-      : [...new Set([...columnState, ...filteredColumns.map((col) => col.column)])];
+      : [
+          ...new Set([
+            ...columnState,
+            ...filteredColumns.map((col) => {
+              if (col.location === V1LocationType.RUNMETADATA) return formatColumnKey(col);
+              return col.column;
+            }),
```

Comment:
`formatColumnKey` already does this check
```suggestion
            ...filteredColumns.map((col) => formatColumnKey(col)),
```